src/OSPSuite.SimModelNative/src/IfFormula.cpp

```diff
@@ -163,15 +169,19 @@ void IfFormula::DE_Jacobian (double * * jacobian, const double * y, const double
 	assert(m_ThenStatement != NULL);
 	assert(m_ElseStatement != NULL);
 
-	if (m_IfStatement->DE_Compute(y, time, USE_SCALEFACTOR) == 1)
+	auto ifStatementValue = m_IfStatement->DE_Compute(y, time, USE_SCALEFACTOR);
+	if (isnan(ifStatementValue))
+		return; //TODO is this ok? or should we throw an error instead?
```

Comment:
here (and in other functions) it's the question for me, what should happen if NaN appears in the jacobian function:
* do nothing?
* throw an error?
* set the corresponding Jac value to NaN?

Reply:
all right,. setting to NaN is also a good option...hum
@abdullahhamadeh Any gut feel here? (This is about what do we do when we have an ? operation and the if condition evaluates to NaN ).

Reply:
Actually I think it's OK AS IS (= "do nothing").
Because if we have an NaN in the conditional statement - the corresponding RHS value will be NaN anyways, which will produce the error immediately after the current RHS function evaluation is finished.
Setting additionally some Jac-Values to NaN will then have no additional effect.